src/lib/viewers/archive/__tests__/ArchiveExplorer-test-react.js

```diff
@@ -65,218 +78,124 @@ describe('lib/viewers/archive/ArchiveExplorer', () => {
     });
 
-    describe('handleItemClick()', () => {
-        test('should set state when handleItemClick() is called', () => {
-            const component = getComponent({ filename, itemCollection: data });
-
-            component.instance().handleItemClick({ fullPath: 'test/subfolder/' });
+    describe('Subfolders', () => {
+        test('should render nested content on folder item click', async () => {
+            renderView({ filename, itemCollection: data });
 
-            expect(component.state().fullPath).toBe('test/subfolder/');
-            expect(component.state().view).toBe(VIEWS.VIEW_FOLDER);
-            expect(component.state().searchQuery).toBe('');
-        });
-    });
+            await userEvent.click(screen.getByText('test'));
 
-    describe('handleBreadcrumbClick()', () => {
-        test('should set state when handleBreadcrumbClick() is called', () => {
-            const component = getComponent({ filename, itemCollection: data });
+            expect(screen.getByText('subfolder')).toBeInTheDocument();
+            expect(screen.getByText('pdf-level-1.pdf')).toBeInTheDocument();
 
-            component.instance().handleBreadcrumbClick('test/subfolder/');
+            await userEvent.click(screen.getByText('subfolder'));
 
-            expect(component.state().fullPath).toBe('test/subfolder/');
+            expect(screen.getByText('test-level-2.jpg')).toBeInTheDocument();
         });
     });
 
-    describe('getRowData()', () => {
-        test('should return correct row data', () => {
-            const component = getComponent({ filename, itemCollection: data });
-
-            const rowData = component.instance().getRowData(data)({ index: 0 });
-
-            const { KEY_NAME, KEY_MODIFIED_AT, KEY_SIZE } = TABLE_COLUMNS;
-            const { absolute_path: fullPath, modified_at: modifiedAt, name, size, type, ...rest } = data[0];
-
-            expect(rowData).toEqual({
-                [KEY_NAME]: {
-                    fullPath,
-                    isExternal: false,
-                    name,
-                    type,
-                    dataAttributes: {
-                        'data-resin-target': type,
-                    },
-                },
-                [KEY_MODIFIED_AT]: modifiedAt,
-                [KEY_SIZE]: type === 'folder' ? null : size,
-                ...rest,
-            });
-        });
-    });
+    describe('Breadcrumbs', () => {
+        test('should render breadcrumbs correctly', async () => {
+            renderView({ filename, itemCollection: data });
 
-    describe('getItemList()', () => {
-        test('should return correct item list', () => {
-            const component = getComponent({ filename, itemCollection: data });
+            await userEvent.click(screen.getByText('test'));
 
-            let itemList = component.instance().getItemList(data, ROOT_FOLDER);
+            expect(within(screen.getByLabelText('Breadcrumb')).getByText('test.zip')).toBeInTheDocument();
```

Comment:
:thought_balloon: Isn't it enough to just search for this element in the document/screen which would fail if it does't exist?

Reply:
Do you mean this 👇  @rafalmaksymiuk ?

```suggestion
            expect(screen.getByText('test.zip')).toBeInTheDocument();
```

If yes, I think this shortened version is much more readable.

Reply:
Rather simple `within(screen.getByLabelText('Breadcrumb')).getByText('test.zip')` which would serve same purpose, but then there is good argument here about the intention:
https://stackoverflow.com/questions/73281448/why-we-have-to-use-tobeinthedocument-in-testing-library

Reply:
@rafalmaksymiuk yes, I referred to the same recommendation